### DISCUSSION TITLE
Load PDF

### DISCUSSION THREAD

**Discussion by svoboda200786** [fabricjs/fabric.js#7522] [Q&A] [answered]
Is there any way to load PDF, like we load JSON? I mean, PDF should be editable.

**Comment by asturur** [comment]
hi @svoboda200786 that would require a PDF parser that i don't have.
Do you have one?

**Comment by ShaMan123** [comment, marked as answer]
https://github.com/mozilla/pdf.js

**Reply by ShaMan123** [reply]
I think I had something working with pdf import...
Do we want this inside fabric?

**Reply by ShaMan123** [reply]
Take a look 
https://codesandbox.io/s/fabric-app-forked-oguic

**Reply by asturur** [reply]
pdf import would be cool, but definitely an external plugin.
pdf.js does render it on a canvas, but if we need to look into that, we want to break the pdf in paths/text and render it as fabric object grouped.
Anything less than that, can be explained with you code sandbox in a tutorial and is great as it is.

**Reply by asturur** [reply]
There is a fork of pdf.js that returns you a json with information. i found it 3-4 days ago after googling after answering this thread

**Reply by ShaMan123** [reply]
So we wait for a proper way to create plugins?

**Reply by ShaMan123** [reply]
Wow if there's path data and we create a group this becomes limitless in terms of editing the pdf.
Cool

**Reply by asturur** [reply]
Yes PDF share similarities with canvas in how the transform work.
It is a bit different for other things but is definitely importable. It needs a parser that can read the PDF and extract the informations in a way that can be converted in object.
Is a cool feature many people asked about, but as plain pages/images is fine as a tutorial.

**Reply by ShaMan123** [reply]
can you post here the fork?

**Reply by ShaMan123** [reply]
https://github.com/fabricjs/fabricjs.com/pull/325

**Reply by asturur** [reply]
The demo does not work :(

**Reply by ShaMan123** [reply]
strange.

**Reply by ShaMan123** [reply]
It doesn't work with the local server, non of the codepens do, but look
https://codepen.io/shaman123/pen/YzEWxpY

**Reply by asturur** [reply]
Yes weird i can't really see quickly what is it.
Something is returning 500 and blocking everything else
